invenio_userprofiles/forms.py

```diff
@@ -123,37 +123,42 @@ def populate_obj(self, user):
         super().populate_obj(user)
 
 
-class EmailProfileForm(ProfileForm):
-    """Form to allow editing of email address."""
+def create_email_profile_form(UserProfileForm):
+    """Factory function to create EmailProfileForm inheriting from UserProfileForm."""
+
+    class EmailProfileForm(UserProfileForm):
+        """Form to allow editing of email address."""
+
+        email = StringField(
+            # NOTE: Form field label
+            _("Email address"),
+            filters=[
+                lambda x: x.lower() if x is not None else x,
+            ],
+            validators=[
+                email_required,
+                current_user_email,
+                email_validator,
+                unique_user_email,
+            ],
+        )
 
-    email = StringField(
-        # NOTE: Form field label
-        _("Email address"),
-        filters=[
-            lambda x: x.lower() if x is not None else x,
-        ],
-        validators=[
-            email_required,
-            current_user_email,
-            email_validator,
-            unique_user_email,
-        ],
-    )
+        email_repeat = StringField(
+            # NOTE: Form field label
+            _("Re-enter email address"),
+            # NOTE: Form field help text
+            description=_("Please re-enter your email address."),
```

Comment:
nit: it is nice to tell the user why we expect them to do an action, otherwise the description is almost the same as the field label
```suggestion
            description=_("Please re-enter your email address to make sure the provided address is correct."),
```